diff -PNaur boost_1_46_1.orig/libs/filesystem/v3/src/path.cpp boost_1_46_1/libs/filesystem/v3/src/path.cpp
--- boost_1_46_1.orig/libs/filesystem/v3/src/path.cpp	2011-03-03 10:00:21.000000000 -0800
+++ boost_1_46_1/libs/filesystem/v3/src/path.cpp	2011-04-21 11:42:27.000000000 -0700
@@ -207,7 +207,7 @@
 # ifdef BOOST_WINDOWS_API
   path & path::make_preferred()
   {
-    std::replace(m_pathname.begin(), m_pathname.end(), L'\\', L'/');
+    std::replace(m_pathname.begin(), m_pathname.end(), L'/', L'\\');
     return *this;
   }
 # endif
