--- libarchive-2.8.4.orig/libarchive/archive_windows.c	2011-05-09 12:36:15.000000000 -0700
+++ libarchive-2.8.4/libarchive/archive_windows.c	2011-05-09 12:50:06.000000000 -0700
@@ -109,10 +109,13 @@
 static __inline ino_t
 getino(struct ustat *ub)
 {
+	ino_t res;
 	ULARGE_INTEGER ino64;
 	ino64.QuadPart = ub->st_ino;
 	/* I don't know this hashing is correct way */
-	return (ino64.LowPart ^ (ino64.LowPart >> INOSIZE));
+	res = ino64.LowPart & 0xffff;
+	res ^= (ino64.LowPart >> INOSIZE) & 0xffff;
+	return res;
 }
 
 /*
